hopp/tools/design/wind/power_curve_tools.py

```diff
@@ -0,0 +1,174 @@
+    plt.legend()
+    plt.xlabel("Wind Speed [m/s]")
+    plt.show()
+
+
+def pad_power_curve(wind_speed, curve, v_min = 0.0, v_max = 50.0):
+    """Pad curve data with zeros from wind speeds of v_min to v_max.
+
+    Args:
+        wind_speeds_ms (List | np.ndarray): list of wind speeds in m/s
+        curve (List | np.ndarray): curve data at each wind speed in `wind_speeds_ms`. Can be 
+            either Cp, power, or Ct.
+        v_min (float, Optional): wind speed to start curve data at. Defaults to 0.0.
+        v_max (float, Optional): wind speed to end curve data at. Defaults to 50.0.
+
+    Returns:
+       2-element tuple containing
+
+       - **padded_wind_speeds_ms** (list): padded wind speed in m/s starting at v_min and ending at v_max
+       - **padded_curve** (list): padded curve data for wind speeds starting at v_min and ending at v_max
+    """
+    wind_speeds_ms = copy.deepcopy(wind_speed)
+    
+    if isinstance(wind_speeds_ms,list):
+        wind_speeds_ms = np.array(wind_speeds_ms)
+    if isinstance(curve,list):
+        curve = np.array(curve)
+
+    if min(wind_speeds_ms) > v_min:
+        wind_speed_pad = np.arange(v_min,min(wind_speeds_ms),1)
+        wind_speeds_ms = np.concatenate((wind_speed_pad,wind_speeds_ms))
+        curve = np.concatenate((np.zeros(len(wind_speed_pad)),curve))
+
+    if max(wind_speeds_ms) < v_max:
+        wind_speed_pad = np.arange(max(wind_speeds_ms),v_max,1)
+        wind_speeds_ms = np.concatenate((wind_speeds_ms,wind_speed_pad))
+        curve = np.concatenate((curve,np.zeros(len(wind_speed_pad))))
+    return list(wind_speeds_ms), list(curve)
+
+def calculate_cp_from_power(wind_speeds_ms, power_curve_kw, rotor_diameter, air_density = 1.225):
+    """Calculate power coefficient curve (Cp) from power curve.
+
+    Args:
+        wind_speeds_ms (List | np.ndarray): list of wind speeds in m/s
+        power_curve_kw (List | np.ndarray): turbine power (in kW) at each wind speed in `wind_speeds_ms`
+        rotor_diameter (float): rotor diameter of the turbine in meters.
+        air_density (float, Optional): Air density assumed for power-curve calculations in kg/m3.
+             Defaults to 1.225.
+
+    Raises:
+        UserWarning: if `wind_speeds_ms` and `power_curve_kw` are different lengths.
+
+    Returns:
+        list: power curve coefficients (Cp) at each wind speed in `wind_speeds_ms`
+    """
+
+    if len(wind_speeds_ms) != len(power_curve_kw):
+        raise UserWarning("The length of the wind speed and power vectors must be the same")
+    rotor_area = np.pi*((rotor_diameter/2)**2)
+    if isinstance(wind_speeds_ms, list):
+        wind_speeds_ms = np.array(wind_speeds_ms)
+    if isinstance(power_curve_kw, list):
+        power_curve_kw = np.array(power_curve_kw)
+    
+    # power available in the wind (kW)
+    p_wind = 0.5*air_density*rotor_area*(wind_speeds_ms**3)/1e3
+    cp = power_curve_kw/p_wind
+    cp = np.where(cp < 0, 0, cp)
```

Comment:
I would just add a `.tolist()` at the end instead of converting in the return statement.